tests/components/lastfm/test_sensor.py

```diff
@@ -0,0 +1,87 @@
+"""Tests for the lastfm sensor."""
+from unittest.mock import patch
+
+from pylast import Track
+import pytest
+
+from homeassistant.components import sensor
+from homeassistant.components.lastfm.sensor import STATE_NOT_SCROBBLING
+from homeassistant.setup import async_setup_component
+
+
+class MockUser:
+    """Mock user object for pylast."""
+
+    def __init__(self, now_playing_result):
+        """Initialize the mock."""
+        self._now_playing_result = now_playing_result
+
+    def get_playcount(self):
+        """Get mock play count."""
+        return 1
+
+    def get_image(self):
+        """Get mock play count."""
+        pass
+
+    def get_recent_tracks(self, limit):
+        """Get mock recent tracks."""
+        return []
+
+    def get_top_tracks(self, limit):
+        """Get mock top tracks."""
+        return []
+
+    def get_now_playing(self):
+        """Get mock now playing."""
+        return self._now_playing_result
+
+
+@pytest.fixture(name="get_user")
+def lastfm_get_user_fixture():
+    """Create fixture for get_user."""
+    with patch("pylast.LastFMNetwork.get_user") as get_user:
+        yield get_user
+
+
+@pytest.fixture(name="lastfm_network")
+def lastfm_network_fixture():
+    """Create fixture for _Network."""
+    with patch("pylast._Network") as lastfm_network:
```

Comment:
We want to patch the `LastFMNetwork` class which is the child of the parent here.

Reply:
Should I be patching `pylast.LastFMNetwork` or the one from the component?

Reply:
What should I be mocking on this class btw? It seems to have broken the tests for me :(

Reply:
Since the whole module or package is imported we can patch the short path directly. If we do `from ... import ...` we would have to patch the long path that starts with `"homeassistant.components."`.

Reply:
I think I figured it out - although I am still not sure I should be patching `pylast.LAstFMNetwork` It fails otherwise though.